packages/block-editor/src/components/inserter/menu.js

```diff
@@ -66,18 +66,21 @@ function InserterMenu(
 			insertionIndex: __experimentalInsertionIndex,
 			shouldFocusBlock,
 		} );
-	const { showPatterns } = useSelect(
+	const { showBlocks, showPatterns } = useSelect(
 		( select ) => {
-			const { hasAllowedPatterns } = unlock( select( blockEditorStore ) );
+			const { hasAllowedPatterns, __unstableGetEditorMode } = unlock(
+				select( blockEditorStore )
+			);
 			return {
+				showBlocks: __unstableGetEditorMode() !== 'zoom-out',
 				showPatterns: hasAllowedPatterns( destinationRootClientId ),
 			};
 		},
 		[ destinationRootClientId ]
 	);
 
 	const mediaCategories = useMediaCategories( destinationRootClientId );
-	const showMedia = mediaCategories.length > 0;
+	const showMedia = mediaCategories.length > 0 && showBlocks;
```

Comment:
@priethor If we're going to use this here I would suggest we rename the variable to `isZoomedOutMode` so its clearer what is going on.

Reply:
Done in [dce6cca](https://github.com/WordPress/gutenberg/pull/59245/commits/dce6cca2c71a083bc56dd4de7778f56269f2de3e)